Contributors-Guide/GlusterFS-Release-process.md

```diff
@@ -1,28 +1,116 @@
+
+Minor releases will have guaranteed backwards compatibilty with earlier minor releases of the same branch.
+
+- LTM and STM state alternates with every release.
+- LTM and STM releases differ in the period of updates provided.
+
+### Long Term Maintenance (LTM) releases
+An LTM release reaches EOL when a LTM+2 release is done, 1 year later. LTM releases may get monthly minor releases for bug-fixes.
+
+### Short Term Maintenance (STM) releases
+STM releases are only supported until the next major release. STM releases may get monthly updates for bug fixes.
+
+The discussions around release schedules and lifecycles happened in the following mail threads on the GlusterFS maintainers mailing list.
+
+- https://www.gluster.org/pipermail/maintainers/2016-June/000892.html
+- https://www.gluster.org/pipermail/maintainers/2016-August/001174.html
+
+
+## GlusterFS major release window
+Each GlusterFS major release has a 3 month release window, in which changes get merged. The 3 month release window is split into two phases.
+
+1. A Open phase, where all changes get merged
+1. A Stability phase, where only changes that stabilize the release get merged.
+
+The first 2 months of a release window will be the Open phase, and the last month will be the stability phase.
+
+The release engineer (or team doing the release) is responsible for messaging.
+
+#### Open phase
+Any changes are accepted during this phase. New features that are introduced in this phase, need to be capable of being selectively built. All changes in the master branch are automatically incuded in the next release.
+
+All changes will be accepted during the Open phase. The changes have a few requirements,
+
+- a change fixing a bug SHOULD have public test case
+- a change introducing a new feature MUST have a disable switch that can disable the feature during a build
+
+
+#### Stability phase
+This phase is used to stabilize any new features introduced in the open phase, or general bug fixes for already existing features. A new release-<version> branch is created, all changes need to be sent to the master branch before getting backported to the new release branch. [kshlm: I prefer that the release gets tagged and branched at the end of the stability phase. This reduces the need for backports.]
+
+No new features will be merged in this phase.
 
-Notify packagers
-----------------
+At the end of this phase, any new feature introduced during the Open phase, that hasn't been declared stable, will be disabled, if possible removed to prevent confusion and set clear expectations towards users and developers.
 
-Notify the packagers that we need packages created. Provide the link to the
-source tarball from the Jenkins release job to the [packagers
-mailinglist](mailto:packaging@gluster.org). A list of the people involved in
-the package maintenance for the different distributions is in the `MAINTAINERS`
-file in the sources.
+The branching of the release happens at the end of this phase.
```

Comment:
I'll change this, but I'll give the reason why I thought it would be better to branch at the end (or as close to the end as possible).

When we approach a release date, we get a lot of stability fixes. But as we have branched off already, each of this fix needs to be pushed twice, reviewed twice, verified twice, and needs to be tracked carefully so that it lands on both master and the branch. This IMO is a lot of effort close to release for all (maintainers, developers and the infra), which could be avoided.

Also, IMO blocking patches on master for a small period doesn't hurt a lot. In this period I expect maintainers and developers to be focused on getting the release stable, which should lead to generally lower feature activity on master. 

Would a compromise work here? We branch in the middle of the stability phase. The branch point would be RC1 of the release. For the first 15 days of the stability phase, the master branch only accepts stability fixes. This is not particularly long. All contributors can focus on getting the release stable at this time. Once branched, maintainers are expected to verify and validate the stability of their components. Other developers are now free to get back to developing features on master.